File: match-service/src/main/java/fact/it/matchservice/dto/MatchResponse.java
package fact.it.matchservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class MatchResponse {
    private String matchCode;
    private TeamResponse homeTeam;
    private TeamResponse awayTeam;
    private LocalDateTime dateTime;
    private String location;
    private Integer homeTeamScore;
    private Integer awayTeamScore;
}

File: soccer-frontend/src/dto/MatchResponse.ts
import { TeamResponse } from "./TeamResponse";

export interface MatchResponse {
    matchCode: string;
    homeTeam: TeamResponse;
    awayTeam: TeamResponse;
    dateTime: string; // Using string for LocalDateTime (ISO-8601 format)
    location: string;
    homeTeamScore: number;
    awayTeamScore: number;
  }